monai/apps/auto3d/data_analyzer.py

```diff
@@ -0,0 +1,670 @@
+    problem is supported, so the user needs to provide paths to the image and label files. Also, label
+    data format is preferred to be (1,H,W,D), with the label index in the first dimension. If it is in
+    onehot format, it will be converted to the preferred format.
+
+    Args:
+        datalist: a Python dictionary storing group, fold, and other information of the medical
+            image dataset, or a string to the JSON file storing the dictionary.
+        dataroot: user's local directory containing the datasets.
+        output_path: path to save the analysis result.
+        average: whether to average the statistical value across different image modalities.
+        do_ccp: apply the connected component algorithm to process the labels/images
+        device: a string specifying hardware (CUDA/CPU) utilized for the operations.
+        worker: number of workers to use for parallel processing.
+        image_only: boolean set to True if the user wants to use the DataAnalyzer on a dataset without labels.
+
+    For example:
+
+    .. code-block:: python
+
+        from monai.apps.auto3d.data_analyzer import DataAnalyzer
+
+        datalist = {
+            "testing": [{"image": "image_003.nii.gz"}],
+            "training": [
+                {"fold": 0, "image": "image_001.nii.gz", "label": "label_001.nii.gz"},
+                {"fold": 0, "image": "image_002.nii.gz", "label": "label_002.nii.gz"},
+                {"fold": 1, "image": "image_001.nii.gz", "label": "label_001.nii.gz"},
+                {"fold": 1, "image": "image_004.nii.gz", "label": "label_004.nii.gz"},
+            ],
+        }
+
+        dataroot = '/datasets' # the directory where you have the image files (nii.gz)
+        DataAnalyzer(datalist, dataroot)
+
+    Note:
+        The module can also be called from the command line interface (CLI) if MONAI is
+        installed with the source code.
+
+    For example:
+
+    .. code-block:: bash
+
+        python monai/apps/auto3d \
+            DataAnalyzer \
+            get_all_case_stats \
+            --datalist="my_datalist.json" \
+            --dataroot="my_dataroot_dir"
+
+    """
+
+    def __init__(
+        self,
+        datalist: Union[str, Dict],
+        dataroot: str = "",
+        output_path: str = "./data_stats.yaml",
+        average: bool = False,
+        do_ccp: bool = True,
+        device: Union[str, torch.device] = "cuda",
+        worker: int = 2,
+        image_only: bool = False,
+    ):
+        """
+        The initializer will load the data and register the functions for data statistics gathering.
+        """
+        if path.isfile(output_path):
+            warnings.warn(f"File {output_path} already exists and will be overwritten.")
+            logger.debug(f"{output_path} will be overwritten by a new datastat.")
+        self.output_path = output_path
+        self.image_only = image_only
+
+        if self.image_only:
+            keys = ["image"]
```

Comment:
we can make the key values optional instead of fixed value. the default values are "image" and "label". and user can use it for different keys in the data list. otherwise we need to let user know the correct keys when the program cannot find them in the data list.